### PR TITLE
Feature: Add ``from_embeddings`` class method to Chroma

### PR DESCRIPTION
**Pull request by l-salewski**
# Description
This PR adds a ``from_embeddings`` class method to the ``Chroma`` vectorstore. This can be handy, if one wants to replace vectorstores that have that method (e.g. ``FAISS``) with ``Chroma``. Additionally, this is useful if one already has pre-computed embeddings and does not want to re-compute them.

The implementation is very simple and just gets the Chroma collection, splits the ``text_embeddings`` into ``texts`` and ``embeddings`` and then calls the ``add_texts`` method. Within ``add_texts`` we need to add a check that only if ``embeddings is None`` we embed the documents to avoid re-embedding.

@rlancemartin and @eyurtsev would be happy to get some feedback!

## Usecase
To use the self-query feature from langchain for my [scholarGPT](https://twitter.com/L_Salewski/status/1673084085474820096) academic chatbot I moved from ``FAISS`` to ``Chroma`` and I wanted a drop-in replacement. 

## Notes
- This feature does not introduce any new dependencies.
- My Twitter handle is [@L_Salewski](https://twitter.com/L_Salewski)

### CODE REVIEW

File: langchain/vectorstores/chroma.py [hunk 1]
```diff
@@ -574,6 +574,51 @@ def from_documents(
...
+            collection_name=collection_name,
+            embedding_function=embedding,
+            persist_directory=persist_directory,
+            client_settings=client_settings,
+            client=client,
+        )
+        # split the text_embeddings into text and embeddings
+        texts = [item[0] for item in text_embeddings]
+        embeddings = [item[1] for item in text_embeddings]
+        chroma_collection.add_texts(
```
Comment (baskaryan) [inline langchain/vectorstores/chroma.py:617]: i don't think add_texts actually takes into account embeddings that are passed in? maybe something worth adding

Comment (l-salewski) [reply langchain/vectorstores/chroma.py:618]: Indeed, thanks for pointing this out @baskaryan! This was actually a copy-paste error, I did add the new parameter in my local version but overlooked it when transfering it to my langchain fork. I updated the PR, let me know what you think!

**Comment by l-salewski** [comment]
Hey @baskaryan, is there anything I can do to improve this PR to get it merged?

**Comment by rlancemartin** [comment]
> Hey @baskaryan, is there anything I can do to improve this PR to get it merged?

I'll have a look today

File: langchain/vectorstores/chroma.py [hunk 2]
```diff
@@ -574,6 +574,52 @@ def from_documents(
...
+        Otherwise, the data will be ephemeral in-memory.
+
+        Args:
+            text_embeddings (List[Tuple[str, List[float]]]): List of tuples with texts
+            and embeddings (list of floats).
+            metadatas (Optional[List[dict]]): List of metadatas. Defaults to None.
+            collection_name (str): Name of the collection to create.
+            persist_directory (Optional[str]): Directory to persist the collection.
+            ids (Optional[List[str]]): List of document IDs. Defaults to None.
+            embedding (Optional[Embeddings]): Embedding function. Defaults to None.
```
Comment (rlancemartin) [inline langchain/vectorstores/chroma.py:602]: Should `embedding` for embedding function be optionally supplied if method requires embeddings to be provided already?

Comment (l-salewski) [reply langchain/vectorstores/chroma.py]: Good point, the ``embedding`` function indeed need to embed the queries later when the vectorstore is used. I pushed a commit to fix this.

**Comment by leo-gan** [comment]
@l-salewski Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

**Comment by l-salewski** [comment]
Hi @leo-gan, unfortunately I am quite busy the upcoming weeks, but I will try to get to it towards the end of october.

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head!

**Pull request closed without merging**